collect-all-transactions/main.go

```diff
@@ -131,6 +163,521 @@ func collectAllTxns(address string) error {
+						data1 := []string{"", "", "", "", "", "",
+							"", "", "", "", "", "", "", "", "",
+							fmt.Sprintf("%v", msg["@type"]),
+							fmt.Sprintf("%v", msg["from_address"]),
+							fmt.Sprintf("%v", msg["to_address"]),
+							amounts[0].Amount,
+						}
+
+						if err = writer.Write(data1); err != nil {
+							return fmt.Errorf("Error in writing data: %s", err.Error())
+						}
+					} else {
+						data = append(data, []string{
+							fmt.Sprintf("%v", msg["@type"]),
+							fmt.Sprintf("%v", msg["from_address"]),
+							fmt.Sprintf("%v", msg["to_address"]),
+							amounts[0].Amount,
+						}...)
+					}
+				}
+
+				if msg["@type"] == "/cosmos.staking.v1beta1.MsgDelegate" {
+					var arrLog []Log
+					amount := msg["amount"]
+					amountBytes, err := json.Marshal(amount)
+					if err != nil {
+						return fmt.Errorf("Error in marshaling Tx: %s", err.Error())
+					}
+
+					var amounts AmountObj
+
+					if err = json.Unmarshal(amountBytes, &amounts); err != nil {
+						return fmt.Errorf("Error in unmarshaling the JSON data: %s", err.Error())
+					}
+
+					if len(messages) > 1 && i != 0 {
+						var rewards string
+
+						if err = json.Unmarshal(logs, &arrLog); err != nil {
+							return fmt.Errorf("Error in unmarshaling the JSON data: %s", err.Error())
+						}
+
+						log := arrLog[i]
+						var found bool
+						var index int
+
+						for lI, e := range log.Events {
+							if e.Type == "transfer" {
+								for _, a := range e.Attributes {
+									if a.Key == "recipient" && a.Value == address {
+										index = lI
+										found = true
+									}
+								}
+							}
+						}
+
+						if found {
+							attrs := log.Events[index]
+							for _, atV := range attrs.Attributes {
+								if atV.Key == "amount" {
+									rewards = atV.Value
+								}
+							}
+						}
+
+						data1 := []string{"", "", "", "", "", "",
+							"", "", "", "", "", "", "", "", "",
+							fmt.Sprintf("%v", msg["@type"]),
+							fmt.Sprintf("%v", msg["delegator_address"]),
+							fmt.Sprintf("%v", msg["validator_address"]),
+							amounts.Amount,
+							"",
+							rewards,
+						}
+
+						if err = writer.Write(data1); err != nil {
+							return fmt.Errorf("Error in writing data: %s", err.Error())
+						}
+					} else {
+						var rewards string
+
+						if err = json.Unmarshal(logs, &arrLog); err != nil {
+							return fmt.Errorf("Error in unmarshaling the JSON data: %s", err.Error())
+						}
+
+						log := arrLog[i]
+						var found bool
+						var index int
+
+						for lI, e := range log.Events {
+							if e.Type == "transfer" {
+								for _, a := range e.Attributes {
+									if a.Key == "recipient" && a.Value == address {
+										index = lI
+										found = true
+									}
+								}
+							}
+						}
+
+						if found {
+							attrs := log.Events[index]
+							for _, atV := range attrs.Attributes {
+								if atV.Key == "amount" {
+									rewards = atV.Value
+								}
+							}
+						}
```

Comment:
You can put this as a utility function and use in multiple places right?

Reply:
Yes, fixed it.